samples/dungeon/app/src/main/java/com/squareup/sample/dungeon/GameSessionWorkflow.kt

```diff
@@ -44,6 +47,7 @@ class GameSessionWorkflow(
     object Loading : State(), Screen
     data class Running(val board: Board) : State()
     data class GameOver(val board: Board) : State()
+    object NewBoard : State()
```

Comment:
The way this is architected, the `DungeonAppWorkflow` is the parent of this `GameSessionWorkflow` and so we don't want to have this as a `State` here where we run `DungeonAppWorkflow` as a child because we are then reproducing this parent and child for each game session.

Instead this should be an `Output` (need to add a type as this currently has `Nothing` as Output type).

Then in `DungeonAppWorkflow` when we run this as a child we would handle the Output and set the state back to `ChoosingBoard`.

That way we won't just keep building an infinite tree of Workflows while re-playing sessions.